zcash_client_backend/src/data_api/testing/transparent.rs

```diff
@@ -13,8 +13,78 @@ use assert_matches::assert_matches;
 use sapling::zip32::ExtendedSpendingKey;
 use zcash_primitives::{
     block::BlockHash,
+    legacy::TransparentAddress,
     transaction::components::{amount::NonNegativeAmount, OutPoint, TxOut},
 };
+use zcash_protocol::local_consensus::LocalNetwork;
+
+use super::TestAccount;
+
+fn check_balance<DSF>(
+    st: &TestState<impl TestCache, <DSF as DataStoreFactory>::DataStore, LocalNetwork>,
+    account: &TestAccount<<DSF as DataStoreFactory>::Account>,
+    taddr: &TransparentAddress,
+    min_confirmations: u32,
+    expected: &Balance,
+) where
+    DSF: DataStoreFactory,
+{
+    // Check the wallet summary returns the expected transparent balance.
+    let summary = st
+        .wallet()
+        .get_wallet_summary(min_confirmations)
+        .unwrap()
+        .unwrap();
+    let balance = summary.account_balances().get(&account.id()).unwrap();
+
+    #[allow(deprecated)]
+    let old_unshielded_value = balance.unshielded();
+    assert_eq!(old_unshielded_value, expected.total());
+    assert_eq!(balance.unshielded_balance(), expected);
+
+    // Check the older APIs for consistency.
+    let mempool_height = st.wallet().chain_height().unwrap().unwrap() + 1;
+    assert_eq!(
+        st.wallet()
+            .get_transparent_balances(account.id(), mempool_height)
+            .unwrap()
+            .get(taddr)
+            .cloned()
+            .unwrap_or(NonNegativeAmount::ZERO),
+        expected.total(),
+    );
+    assert_eq!(
+        st.wallet()
+            .get_spendable_transparent_outputs(taddr, mempool_height, min_confirmations)
+            .unwrap()
+            .into_iter()
+            .map(|utxo| utxo.value())
+            .sum::<Option<NonNegativeAmount>>(),
+        Some(expected.spendable_value()),
+    );
+
+    // we currently treat min_confirmations the same regardless they are 0 (zero confirmations) 
+    // or 1 (one block confirmation). We will check if this assumption holds until it's no 
+    // longer made. If zero and one [`min_confirmations`] are treated differently in the future,
+    // this check should then be removed.
+    if min_confirmations == 0 || min_confirmations == 1 {
+        let min_confirmations: u32 = if min_confirmations == 0 {
```

Comment:
I would just delete this and make the call below using `1 - min_confirmations`.

Reply:
Thanks! I was just trying stuff and got a bit verbose 🙏😅